### PR TITLE
Matlab remove os path dependency

### PR DESCRIPTION
**Pull request by JacobDGrunnet**
When generating a Simulink s-function a number of c-files are compiled using the mex compilation setup and these are linked with a library containing the casadi_ext_fun. The compilation of this library does not use mex compilation, but is rather a system call directly to the compiler. For this to work it is required that the compiler used by mex is added to the OS PATH environment variable.

As we need to have another (incompatible) version of gcc in our OS PATH environment variable acados will use this incompatible gcc version to compile the cadasdi_ext_fun library and the linking will subsequently fail.

### CODE REVIEW

File: interfaces/acados_matlab_octave/ocp_generate_casadi_ext_fun.m [hunk 1]
```diff
@@ -223,6 +226,13 @@ function ocp_generate_casadi_ext_fun(model_struct, opts_struct)
     % build
     compile_command = sprintf('"%s" & %s', msvc_env, build_cmd);
 else % gcc
+    %Read the mex C compiler configuration and extract the location
+    cCompilerConfig = mex.getCompilerConfigurations('C');
```
Comment (JacobDGrunnet) [inline interfaces/acados_matlab_octave/ocp_generate_casadi_ext_fun.m:230]: This command is not available in Octave.

However, it seems that Octave automatically adds a path to compatible gcc version to the PATH environment variable when Octave starts, so it should be possible just to skip all of this if octave is used

**Review by FreyJo — CHANGES_REQUESTED** [review]
Hi, that's good to know.
I have to admit that I don't have a Casadi compatible Octave version on my machine right now, so I can't really test it.
However, that is why we have the Github tests in place.
Can you just add some `if`s to make this work for Octave?
Then the tests should also pass and I will test it for Matlab locally as well.

**Comment by JacobDGrunnet** [comment]
Seems I fixed the remaining bug :D

**Review by FreyJo — APPROVED** [review]
Thanks for fixing the Octave version! Now all tests pass.
I just did some minor edits and will merge this.

**Pull request merged**